docs/reference/wasm.md

```diff
@@ -0,0 +1,241 @@
+
+#### `getDataSource(string name) u32`
+
+Get a handle to a data source.
+
+Parameters:
+- `name` (string): Datasource's name
+
+Return value:
+- (u32) Handle to the data source on success, 0 on error.
+
+#### `dataSourceSubscribe(u32 ds, u32 type, u32 prio, u64 cb)`
+
+Subscribe to events emitted by a data source.
+
+This mechanism requires the wasm module to export a `dsCallback` that is called
+by the host when an event is emitted:
+
+`dsCallback(u64 cbID, u32 ds, u32 data)`
+- `cbID`: Callback ID
+- `ds`: Datasource handle
+- `data`: Data handle
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+- `type` (u32): Subscription type: (0: Data, 1: Array, 2: Packet)
+- `priority` (u32): Priority of the subscription. Lower means higher priority.
+- `cb` (u64): Callback ID passed to `dsCallback`
+
+Return value:
+- 0 on sucess, 1 in case of error.
+
+#### `dataSourceGetField(u32 ds, string name) u32`
+
+Get a field from a datasource
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+- `name`(string): Field's name
+
+Return value:
+- (u32): Field handle on success, 0 on error.
+
+#### `dataSourceAddField(u32 ds, string name, u32 kind) u32`
+
+Add a field to a data source
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+- `name`(string): Field's name
+- `kind` (u32): Field's kind. See values in https://pkg.go.dev/github.com/inspektor-gadget/inspektor-gadget@main/pkg/gadget-service/api#Kind.
+
+Return value:
+- (u32): Field handle on success, 0 on error.
+
+#### `dataSourceNewPacketSingle(u32 ds) u32`
+
+Allocate a packet instance. The returned packet has to be released with
+`dataSourceEmitAndRelease` or `dataSourceRelease`.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+
+Return value:
+- (u32): Packet handle on success, 0 on error
+
+#### `dataSourceNewPacketArray(u32 ds) u32`
+
+Allocate a packet array instance. The returned packet has to be released with
+`dataSourceEmitAndRelease` or `dataSourceRelease`.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+
+Return value:
+- (u32): Packet handle on success, 0 on error
+
+#### `dataSourceEmitAndRelease(u32 ds, u32 data) u32`
+
+Emit and release a packet instance.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+- `data` (u32): Packet handle (as returned by `dataSourceNewPacketSingle`)
+
+Return value:
+- 0 in case of success, 1 otherwise.
+
+#### `dataSourceRelease(u32 ds, u32 data)`
+
+Release a packet instance without sending it.
+
+Parameters:
+- `ds` (u32): Datasource handle (as returned by `getDataSource` or `newDataSource`)
+- `data` (u32): Packet handle (as returned by `dataSourceNewPacketSingle`)
+
+Return value:
+- None
+
+#### `dataArrayNew(d uint32) uint32`
```

Comment:
I think those names are a bit misleading - it sounds like this would create a new array instead of just a data entity based on the type of the array (`dataArrayNewData` would be clearer but inconsistent with the Go API).

I'm fine with not opening that box right now (and it's mostly used internally anyway). 😄